source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/sensor_base.py

```diff
@@ -75,36 +113,63 @@ def set_visibility(self, visible: bool):
             overriding this method is optional.
 
         Args:
-            visible (bool) -- Whether to make instance visible or invisible.
+            visible (bool): Whether to make instance visible or invisible.
         """
-        pass
+        # check camera prim
+        if len(self.count) == 0:
+            raise RuntimeError("Camera prims are None. Please call 'initialize(...)' first.")
+        # set visibility
+        self.view.set_visibilities([visible] * self.count)
 
     """
     Operations
     """
 
     @abstractmethod
-    def spawn(self, parent_prim_path: str):
+    def spawn(self, prim_path: str, *args, **kwargs):
         """Spawns the sensor into the stage.
 
         Args:
-            parent_prim_path (str): The path of the parent prim to attach sensor to.
+            prim_path (str): The path of the prim to attach the sensor to.
         """
         raise NotImplementedError
 
-    @abstractmethod
-    def initialize(self):
-        """Initializes the sensor handles and internal buffers."""
-        raise NotImplementedError
+    def initialize(self, prim_paths_expr: str = None):
+        """Initializes the sensor handles and internal buffers.
+
+        Args:
+            prim_paths_expr (str, optional): The prim path expression for the sensors. Defaults to None.
+        """
+        # Check that view is not None
+        if self._view is None:
+            self._view = XFormPrimView(prim_paths_expr, "sensor_view", reset_xform_properties=True)
+        # Create constant for the number of sensors
+        self._ALL_INDICES = self._backend_utils.resolve_indices(None, self.count, device=self.device)
+        # Current timestamp of animation (in seconds)
+        self._timestamp = self._backend_utils.create_zeros_tensor((self.count,), "float32", self.device)
+        # Timestamp from last update
+        self._timestamp_last_update = self._backend_utils.create_zeros_tensor((self.count,), "float32", self.device)
+        # Frame number when the measurement is taken
+        self._frame = self._backend_utils.create_zeros_tensor((self.count,), "int64", self.device)
+        # Set the initialized flag
+        self._is_initialized = True
 
-    def reset(self):
-        """Resets the sensor internals."""
-        # Set current time
-        self._timestamp = 0.0
-        # Set zero captures
-        self._frame = 0
+    def reset_buffers(self, sensor_ids: Optional[Sequence[int]] = None):
```

Comment:
Shouldn't this method not also called buffer(sensor_ids) to have the fresh data on reset?

Reply:
Also why did you add _buffer to the names? I feel like it was clearer before